app/models/account/states.rb

```diff
@@ -91,17 +93,19 @@ module Account::States
       end
     end
 
+    scope :without_deleted, ->(without = true) { where.has { state != :scheduled_for_deletion } if without }
+
     scope :by_state, ->(state) do
       where(:state => state.to_s) if state.to_s != "all"
     end
 
     scope :deleted_since, ->(value = nil) do
-      scheduled_for_deletion.where.has { deleted_at <= (value || PERIOD_BEFORE_DELETION.ago) }
+      scheduled_for_deletion.where.has { state_changed_at <= (value || PERIOD_BEFORE_DELETION.ago) }
     end
 
     def deletion_date
-      return nil unless deleted_at
-      (deleted_at + PERIOD_BEFORE_DELETION)
+      return nil if state_changed_at.blank? || !scheduled_for_deletion?
```

Comment:
I know it was already there before you even touch it, but no need for the literal `nil` here, right?

Reply:
Right 👍